Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,415 @@
+                {
+                    if (targetList[i] != null)
+                    {
+                        Gizmos.color = Color.red;
+                        Gizmos.DrawLine(this.transform.position, targetList[i].transform.position);
+                    }
+                }
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Function to remove an interactable from the linked list.
+        /// </summary>
+        /// <param name="interactable"></param>
+        public virtual void Removeinteractable(GameObject interactable)
+        {
+            if (interactable != null && interactables.Contains(interactable))
+            {
+                interactables.Remove(interactable);
+            }
+        }
+
+        /// <summary>
+        /// Clear the interactables list and unregister them
+        /// </summary>
+        public virtual void ClearInteractables()
+        {
+            GameObject[] _intList = interactables.ToArray();
+
+            for (int i = 0; i < _intList.Length; i++)
+            {
+                this.Removeinteractable(_intList[i]);
+            }
+        }
+
+        /// <summary>
+        /// Is the game object interactable in our list of interactables
+        /// </summary>
+        /// <param name="interactable"></param>
+        /// <returns></returns>
+        protected bool IsInteractable(GameObject interactable)
+        {
+            return (interactables != null && interactables.Contains(interactable));
+        }
+
+        private void CheckLockFocus( IMixedRealityPointer focuser )
+        {
+            // If our previous selecting focuser isn't the same
+            if (_selectingFocuser != null && _selectingFocuser != focuser)
+            {
+                // If our focus is currently locked, unlock it before moving on
+                if (LockFocus)
+                {
+                    _selectingFocuser.IsFocusLocked = false;
+                }
+            }
+
+            // Set to the new focuser
+            _selectingFocuser = focuser;
+            if (_selectingFocuser != null)
+            {
+                _selectingFocuser.IsFocusLocked = LockFocus;
+            }
+        }
+
+        private void LockFocuser( IMixedRealityPointer focuser )
+        {
+            if (focuser != null)
+            {
+                ReleaseFocuser();
+                _selectingFocuser = focuser;
+                _selectingFocuser.IsFocusLocked = true;
+            }
+        }
+
+        private void ReleaseFocuser()
+        {
+            if (_selectingFocuser != null)
+            {
+                _selectingFocuser.IsFocusLocked = false;
+                _selectingFocuser = null;
+            }
+        }
+
+        public void OnBeforeFocusChange( FocusEventData eventData ) { /*Unused*/ }
+
+        public void OnFocusChanged( FocusEventData eventData )
+        {
+            FocusEventData newData = new FocusEventData( EventSystem.current );
```

Comment:
This doesn't look good. Event data should only be created by the input system.

Reply:
I suppose we can pass the existing FocusEventData into the virtual methods. The original created a new pointer event to pass along. To be honest, I'm not entirely sure why since the new FocusEvent is effectively a clone of the original.